pkg/resourcehandlers/github/github_resource_handler.go

```diff
@@ -335,24 +334,53 @@ func (gh *GitHub) Accept(uri string) bool {
 
 // ResolveNodeSelector recursively adds nodes built from tree entries to node
 // ResolveNodeSelector implements resourcehandlers/ResourceHandler#ResolveNodeSelector
-func (gh *GitHub) ResolveNodeSelector(ctx context.Context, node *api.Node, excludePaths []string, frontMatter map[string]interface{}, excludeFrontMatter map[string]interface{}, depth int32) error {
-	var (
-		rl  *ResourceLocator
-		err error
-	)
-	if rl, err = gh.URLToGitHubLocator(ctx, node.NodeSelector.Path, true); err != nil {
-		return err
+func (gh *GitHub) ResolveNodeSelector(ctx context.Context, node *api.Node, excludePaths []string, frontMatter map[string]interface{}, excludeFrontMatter map[string]interface{}, depth int32) ([]*api.Node, error) {
+	rl, err := gh.URLToGitHubLocator(ctx, node.NodeSelector.Path, true)
+	if err != nil {
+		return nil, err
 	}
-	if rl != nil {
-		// build node subnodes hierarchy from cache (URLToGitHubLocator populates the cache)
-		childResourceLocators := gh.cache.GetSubset(rl.String())
-		if err = buildNodes(node, excludePaths, frontMatter, excludeFrontMatter, depth, childResourceLocators, gh.cache, 0); err != nil {
-			return err
-		}
-		// finally cleanup folder entries from contentSelectors
-		cleanupNodeTree(node)
+	if rl == nil {
+		return nil, nil
 	}
-	return nil
+
+	childResourceLocators := gh.cache.GetSubset(rl.String())
+	childNodes, err := buildNodes(node, excludePaths, frontMatter, excludeFrontMatter, depth, childResourceLocators, gh.cache, 0)
+	if err != nil {
+		return nil, err
+	}
+	// finally cleanup folder entries from contentSelectors
+	for _, child := range childNodes {
+		cleanupNodeTree(child)
+	}
+	if childNodes == nil {
+		return nil, nil
+	}
+
+	return childNodes, nil
+}
+
+// ResolveModule for a give path and return it as a *api.Documentation
+func (gh *GitHub) ResolveModule(ctx context.Context, path string) (*api.Documentation, error) {
+	rl, err := gh.URLToGitHubLocator(ctx, path, true)
+	if err != nil {
+		return nil, err
+	}
+	if rl == nil {
+		return nil, nil
+	}
+	// TODO: In cases where nodesSelector.Path is set to an url poiting to a resource with .md extension, it's
+	// considered as invalid. This is to avoid downloading the resource twice. Contemplate logic that caches
+	// the resource once read for later downloads.
+	if !(rl.Type == Blob || rl.Type == Raw) || urls.Ext(rl.String()) == ".md" {
```

Comment:
In future, we should radically change form using what looks like file extension to detect content type and switch to more heuristics-based method like the one employed by `http.DetectContentType(buffer)`. The negative part is that we need to read content to be able to apply that though.